ciao-controller/internal/datastore/datastore.go

```diff
@@ -2655,21 +2649,39 @@ func (ds *Datastore) GetImage(ID string) (types.Image, error) {
 }
 
 // ResolveImage retrieves an image by name or ID
-func (ds *Datastore) ResolveImage(name string) (types.Image, error) {
-	ds.imageLock.RLock()
-	defer ds.imageLock.RUnlock()
+func (ds *Datastore) ResolveImage(tenantID string, name string) (string, error) {
+	ds.tenantsLock.RLock()
+	defer ds.tenantsLock.RUnlock()
 
-	id, ok := ds.imagesByName[name]
-	if ok {
-		name = id
+	if tenantID != "" && tenantID != "admin" {
+		t, ok := ds.tenants[tenantID]
+		if !ok {
+			return "", fmt.Errorf("Tenant not found: %s", tenantID)
+		}
+
+		for _, id := range t.images {
+			i := ds.images[id]
+			if i.Name == name || i.ID == name {
```

Comment:
I'm wondering about this check.  If you had two different images, one whose id was equal to the name of the second, you would not be able to resolve the second image by name.  Corner case I know, but I thought I'd mention it.

Reply:
Here's the example, I'm thinking about.

Image 1 is created

Id ID1
Name: ID2

Image 2 is created

Id ID2
Name:  hello

This will be allowed by AddImage as image1.ID != image2.ID and image1.Name != image2.Name.  However, it will not be possible to resolve Image2 by its ID.

However, as discussed, since it's no longer possible to specify image IDs when creating an image it's almost impossible that this could happen in practice.

Would it make sense to impose an extra Constraint in AddImage that said that a name cannot be set to an existing id and vice-versa?

Reply:
I think this should be addressed now (and fix problem with updating image and name clash there.)